packages/webpack/src/builder.js

```diff
@@ -216,11 +216,22 @@ export class WebpackBundler {
       )
     )
 
-    // Inject to renderer instance
-    if (server) {
-      server.devMiddleware = this.devMiddleware
-      server.hotMiddleware = this.hotMiddleware
+    // Register devMiddleware on server
+    await this.buildContext.nuxt.callHook('server:devMiddleware', this.middleware)
+  }
+
+  async middleware(req, res, next) {
+    const name = req.modernMode ? 'modern' : 'client'
+
+    if (this.devMiddleware && this.devMiddleware[name]) {
+      await this.devMiddleware[name](req, res)
+    }
+
+    if (this.hotMiddleware && this.hotMiddleware[name]) {
+      await this.hotMiddleware[name](req, res)
```

Comment:
Is the order meaningful here? If not, could we get a perf bonus when using `Promise.all` here?

Reply:
Middleware are sequential by design. Even if it works we are doing an anti pattern. Also only one of them matches a request so the difference. Combining this with the fact that node is single threaded, there should be no difference.